### PR TITLE
signup post route

### PR DESCRIPTION
**Pull request by mossa-Sammer**
hash password with tests and make the database adduser query

### CODE REVIEW

File: server/tests/utils/index.js [hunk 1]
```diff
@@ -1,6 +1,19 @@
 const tape = require('tape');
 
+const { hashPassowrd, comparePassword } = require('../../controllers/utils/password');
+
 tape('Simple pass test', (t) => {
   t.equal(1, 1, 'one should equal one');
   t.end();
 });
```
Comment (hashemt) [inline server/tests/utils/index.js]: delete simple test

File: server/controllers/index.js [hunk 2]
```diff
@@ -1,8 +1,10 @@
 const express = require('express');
```
Comment (FadiAlamassi) [inline server/controllers/index.js]: add a new line after third-party module

File: server/tests/queries/signupQuery.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+const tape = require('tape');
```
Comment (FadiAlamassi) [inline server/tests/queries/signupQuery.js]: add a new line after the core module

File: server/tests/utils/index.js [hunk 4]
```diff
@@ -1,6 +1,19 @@
 const tape = require('tape');
```
Comment (FadiAlamassi) [inline server/tests/utils/index.js]: add a new file for your test and require it here

File: server/tests/utils/index.js [hunk 5]
```diff
@@ -1,6 +1,19 @@
...
 });
+
+tape('type of algorithim used in hash passwords function', (t) => {
+  hashPassowrd('mmm123')
+    .then((hashed) => {
+      t.equal(hashed.substring(1, 3), '2b', 'should be 2b');
+      t.end();
+    })
+    .catch((err) => {
+      t.equal(err, null, 'sholudn\'t be an error');
```
Comment (FadiAlamassi) [inline server/tests/utils/index.js]: add t.end() after every test

File: server/controllers/routes/signup.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
+const { hashPassowrd } = require('../utils/password');
+const addUser = require('../../db/queries/addUser');
+const addHandyman = require('../../db/queries/addHandyman');
+const { handymanSchema } = require('../validationSchemas/handymanSchema');
+const { clientSchema } = require('../validationSchemas/clientSchema');
+const { createToken } = require('../utils/cookie');
```
Comment (FadiAlamassi) [inline server/controllers/routes/signup.js]:
```suggestion

const { hashPassowrd } = require('../utils/password');
const addUser = require('../../db/queries/addUser');
const addHandyman = require('../../db/queries/addHandyman');
const { handymanSchema } = require('../validationSchemas/handymanSchema');
const { clientSchema } = require('../validationSchemas/clientSchema');
const { createToken } = require('../utils/cookie');
```
use one export format when you export any file

File: server/controllers/utils/cookie.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
+const jwt = require('jsonwebtoken');
+require('dotenv').config();
```
Comment (AbdallahAmmar96) [inline server/controllers/utils/cookie.js]: delete dotenv from here

File: server/db/config/fakeData.sql [hunk 8]
```diff
@@ -11,12 +11,12 @@ INSERT INTO handyman (handyman_id, job_title,hour_rate,description)
 VALUES (1,1,15,'good builder');
 
 INSERT INTO jobs (client_id,handyman_id,description,dead_line,price,status,message,street,building_number,flat_number) 
-VALUES (2,1,'build the house','12-12-2020',15,'pending','build my house','omar el mukhtar','123','12');
+VALUES (2,1,'build the house','2020-12-12',15,'pending','build my house','omar el mukhtar','123','12');
```
Comment (AbdallahAmmar96) [inline server/db/config/fakeData.sql]: pull the master and remove this file from P.R

File: server/db/queries/addHandyman.js [hunk 9]
```diff
@@ -0,0 +1,16 @@
+const connection = require('../config/connection');
+
+const addHandyman = ({
+  id,
+  jobTitle,
+  hourRate,
+  description,
+}) => {
+  const sql = {
+    text: 'INSERT INTO handyman (handyman_id,job_title,hour_rate,description) VALUES ($1,$2,$3,$4) returning *',
```
Comment (AbdallahAmmar96) [inline server/db/queries/addHandyman.js]:
```suggestion
    text: 'INSERT INTO handyman (handyman_id, job_title, hour_rate, description) VALUES ($1, $2, $3, $4) returning *',
```

File: server/controllers/validationSchemas/handymanSchema.js [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+  username: yup.string().min(3).required(),
+  email: yup.string().email().required(),
+  password: yup.string().min(8).required(),
+  confirmPassword: yup.string().min(8).required(),
+  phone: yup.number().positive().min(10).required(),
+  country: yup.string().min(3).required(),
+  city: yup.string().min(3).required(),
+  jobTitle: yup.number().required(),
+  hourRate: yup.number().required(),
+  description: yup.string().min(10).required(),
```
Comment (FadiAlamassi) [inline server/controllers/validationSchemas/handymanSchema.js]: add isHandyman to the schema

File: server/db/queries/addUser.js [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+  username,
+  email,
+  phone,
+  password,
+  country,
+  city,
+  isHandyman,
+}) => {
+  const sql = {
+    text: 'INSERT INTO users (username,email,phone,password,country,city,is_handyman) VALUES ($1,$2,$3,$4,$5,$6,$7) returning *',
```
Comment (FadiAlamassi) [inline server/db/queries/addUser.js]: spaces after each value

**Pull request merged**